website/source/docs/guides/kubernetes-reference.html.md

```diff
@@ -0,0 +1,184 @@
+
+### Consul Datacenter with a Kubernetes Cluster
+
+To use an existing Consul cluster to manage services in Kubernetes, Consul
+clients can be deployed within the Kubernetes cluster. This will also allow the
+Kubernetes services to be synced to Consul. This design allows Consul tools
+such as envconsul, consul-template, and more to work on Kubernetes. It will
+also register each Kubernetes node with the Consul catalog for full visibility
+into your infrastructure.
+
+![Reference Diagram](/assets/images/k8s-cluster-consul-datacenter.png "Consul and Kubernetes Reference Diagram")
+
+This type of deployment in Kubernetes can also be set up with the official Helm
+chart.
+
+
+### Multiple Consul Clusters with a Kubernetes Cluster
+
+Consul clusters in different datacenters running the same service can be joined
+by WAN links. The clusters can operate independently and only communicate over
+the WAN. This type datacenter design is detailed in the [Reference Architecture
+guide](/advanced/day-1-operations/reference-architecture#multiple-datacenters).
+In this setup, you can have a Consul cluster running outside of Kubernetes and
+a Consul cluster running inside of Kubernetes. 
+
+### Catalog Sync
+
+To use catalog sync, you must enable it in the [Helm
+chart](https://www.consul.io/docs/platform/k8s/helm.html#v-synccatalog).
+Catalog sync allows you to sync services between Consul and Kubernetes. The
+sync can be unidirectional in either direction or bidirectional. Read the
+[documentation](https://www.consul.io/docs/platform/k8s/service-sync.html) to
+learn more about the configuration. 
+
+Services synced from Kubernetes to Consul will be discoverable, like any other
+service within the Consul datacenter. Read more in the [network
+connectivity](#networking-connectivity) section to learn more about related
+Kubernetes configuration. Services synced from Consul to Kubernetes will be
+discoverable with the built-in Kubernetes DNS once a [Consul stub
+domain](https://www.consul.io/docs/platform/k8s/dns.html) is deployed. When
+bidirectional catalog sync is enabled, it will behave like the two
+unidirectional setups. 
+
+## Networking Connectivity 
+
+When running Consul inside Kubernetes as a pod, the Consul servers will be
+automatically configured with the appropriate addresses. However, when running
+Consul servers outside of the Kubernetes cluster and clients inside Kubernetes
+as pods, there are additional [networking
+considerations](/consul/advanced/day-1-operations/reference-architecture#network-connectivity).
+
+### Network Connectivity for Services
+
+When using Consul catalog sync, to sync Kubernetes services to Consul, you will
+need to ensure the Kubernetes services are supported [service
+types](https://www.consul.io/docs/platform/k8s/service-sync.html#kubernetes-service-types)
+and configure correctly in Kubernetes. If the service is configured correctly,
+it will be discoverable by Consul like any other service in the datacenter. 
```

Comment:
This is technically correct-- k8s services will be discoverable, but one of the things missing here is that external Consul services will not be able to connect to these k8s services unless those k8s services are routable from outside the k8s cluster. Certain types of k8s services do this automatically, but not all of them (i.e. ClusterIP).

There needs to be a warning about this-- that the user has figured out the required networking to connect to things running in k8s.

Reply:
Add a warning box: "~> Warning: You are responsible for ensuring that external services can communicate with services deployed in the Kubernetes cluster. For example, `ClusterIP` type services may not be directly accessible by IP address from outside the Kubernetes cluster for some Kubernetes configurations. "